src/Http/Http.Results/src/AcceptedAtRouteHttpResult.cs

```diff
@@ -44,13 +55,21 @@ public AcceptedAtRouteResult(object? routeValues, object? value)
     /// <summary>
     /// Gets the route data to use for generating the URL.
     /// </summary>
-    public RouteValueDictionary RouteValues { get; }
+    public RouteValueDictionary? RouteValues { get; }
+
+    /// <summary>
+    /// Gets the HTTP status code.
+    /// </summary>
+    public int StatusCode => StatusCodes.Status202Accepted;
+
+    /// <inheritdoc/>
+    public Task ExecuteAsync(HttpContext httpContext)
+        => HttpResultsWriter.WriteResultAsJsonAsync(httpContext, Value, StatusCode, configureResponseHeader: ConfigureResponseHeaders);
```

Comment:
Can we get remove the the `configureResponseHeader` param from `HttpResultsWriter.WriteResultAsJsonAsync`? I'm tempted to remove the `statusCode` param too.

```suggestion
    public Task ExecuteAsync(HttpContext httpContext)
    {
        httpContext.Response.StatusCode = StatusCode;
        ConfigureResponseHeaders(httpContext);
        HttpResultsWriter.WriteResultAsJsonAsync(httpContext, Value);
    }
```

Reply:
I removed the `configureResponseHeader` but I did not find a lot of benefit in removing `statusCode`, I can do if you really prefer but for now, I am keeping.

I also removed the `ApplyProblemDetailsDefaults` from `WriteResultAsJsonAsync` and did it as part of the result ctors.